packages/oauth-adapters/test/oauthAuthenticationAdapter.test.ts

```diff
@@ -28,115 +31,161 @@ describe('test oauth request provider', () => {
       expiry: BigInt(Date.now()),
     };
     const authenticationProvider = requestAuthenticationProvider(oAuthToken);
-    let context: HttpContext = { request, response };
-    const handler = authenticationProvider(true);
-    const interceptor = [handler];
-    const client = async (req) => {
-      return { request: req, response };
-    };
-    const executor = callHttpInterceptors(interceptor, client);
-    context = await executor(request, undefined);
-    return expect(context.request.headers).toEqual({
+    return await executeAndExpect(authenticationProvider(true), {
       authorization: 'Bearer 1f12495f1a1ad9066b51fb3b4e456aee',
     });
   });
 
-  it('should test oauth request provider with disabled authentication', async () => {
-    const response: HttpResponse = {
-      statusCode: 200,
-      body: 'testBody',
-      headers: { 'test-header': 'test-value' },
-    };
-
-    const request: HttpRequest = {
-      method: 'GET',
-      url: 'http://apimatic.hopto.org:3000/test/requestBuilder',
-    };
-
-    const oAuthToken = {
+  it('should pass with valid token + authProvider + updateCallback', async () => {
+    const oAuthToken: OAuthToken = {
       accessToken: '1f12495f1a1ad9066b51fb3b4e456aee',
       tokenType: 'Bearer',
       expiresIn: BigInt(100000),
       scope: '[products, orders]',
       expiry: BigInt(Date.now()),
     };
-    const authenticationProvider = requestAuthenticationProvider(oAuthToken);
-    let context: HttpContext = { request, response };
-    const handler = authenticationProvider(false);
-    const interceptor = [handler];
-    const client = async (req) => {
-      return { request: req, response };
-    };
-    const executor = callHttpInterceptors(interceptor, client);
-    context = await executor(request, undefined);
-    return expect(context.request.headers).toBeUndefined();
+    const authenticationProvider = requestAuthenticationProvider(
+      oAuthToken,
+      (token: OAuthToken | undefined) => {
+        // return an invalid token if accessed from provider
+        if (token === undefined) {
+          return Promise.resolve({
+            accessToken: 'Invalid',
+            tokenType: 'Bearer',
+          });
+        }
+        return Promise.resolve({
+          ...token,
+          accessToken: 'Invalid',
+        });
+      },
+      (_: OAuthToken) => {
+        // fail if token gets updated
+        expect(true).toBe(false);
+      }
```

Comment:
This looks like a very hacky way to test whether callbacks were called. Instead of doing this, use Jest create mock functions for callbacks and then test whether those callback had been called. More found here: https://jestjs.io/docs/mock-functions#using-a-mock-function

Please ensure any tests that require function mocking are updated to use Jest's function mocking and assertions capability.

Reply:
Its a great suggestion, I have incorporated it!